src/local-mode.ts

```diff
@@ -1,12 +1,59 @@
 import * as Bluebird from 'bluebird';
 import * as Docker from 'dockerode';
 import * as _ from 'lodash';
-
 import Config from './config';
 import Database from './db';
+import log from './lib/supervisor-console';
 import { Logger } from './logger';
 
-import log from './lib/supervisor-console';
+// EngineSnapshot represents a list of containers, images, volumens, and networks present on the engine.
+// A snapshot is taken before entering local mode in order to perform cleanup when we exit local mode.
+export class EngineSnapshot {
+	constructor(
+		public readonly containers: string[],
+		public readonly images: string[],
+		public readonly volumes: string[],
+		public readonly networks: string[],
+	) {}
+
+	public static fromJSON(json: string): EngineSnapshot {
```

Comment:
Some cursory validation here would be good I think, fs corruption is something we have to deal with constantly, so any logs detailing that something read from the database is wrong definitely helps when debugging.

Reply:
Good point. Let me check how error here works, and how it's logged...

Reply:
There is no such option :(
Do you know if we can configure `prettify` to do this?

Reply:
In 5de1410e13fa0b3905000f6ab58201d5536a672b, I added code to ensure we log bad input when parsing errors happen. It basically yields to skipping local mode leave changes - see tests in that commit.

Is it good enough for you?

Reply:
That's great, works for me, thanks @roman-mazur